### PR TITLE
[expo-ads-admob] Upgrade underlying native SDKs

### LINKED ISSUE #5242: ExpoKit SDK 34 is not compatible with latest react-native-firebase 5.5.6 due to Google-Mobile-Ads-SDK semver violation
tag ```ios/2.12.3```
Get exception:
```
Property 'customRewardString' not found on object of type 'GADRewardBasedVideoAd *'
```
The issue is caused by the library for AdMob being a lower version (Google-Mobile-Ads-SDK v7.22.0)

### LINKED ISSUE #5330: [ios13] Upgrade Google-Mobile-Ads-SDK to 7.44 or higher
See https://github.com/expo/expo/issues/5242. Basically firebase-ios-sdk 6.3.0 crashes on iOS 13 and 6.5.0+ doesn't. firebase-ios-sdk@6.5.0+ depends on Google-Mobile-Ads-SDK 7.44+. In Expo(Kit) we use Google-Mobile-Ads-SDK 7.22.

The issue is that there appears to be a semver-breaking change between Google-Mobile-Ads-SDK 7.22 and 7.44, so if an app uses 7.44 there is a runtime error as described in the linked issue.

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Fixes https://github.com/expo/expo/issues/5242 and closes https://github.com/expo/expo/issues/5330.

# How

Upgraded underlying native SDKs on both iOS and Android. Added required app IDs to `AndroidManifest` and `Info.plist`. Added support for `googleMobileAdsAppId` to XDL and Universe. Added documentation around `googleMobileAdsAppId`.

x-refs: https://github.com/expo/expo-cli/pull/961, https://github.com/expo/universe/pull/3660

# Test Plan

Expo client builds with these changes on both platforms.

### CODE REVIEW

**Review by ide — CHANGES_REQUESTED** [review]

File: template-files/android/AndroidManifest.xml [hunk 1]
```diff
@@ -359,6 +359,13 @@
       android:value="${GOOGLE_MAPS_API_KEY}"/>
     <!-- END GOOGLE MAPS CONFIG -->
 
+    <!-- ADD GOOGLE MOBILE ADS CONFIG HERE -->
+    <!-- BEGIN GOOGLE MOBILE ADS CONFIG -->
+    <meta-data
+      android:name="com.google.android.gms.ads.APPLICATION_ID"
+      android:value="ca-app-pub-3940256099942544~3347511713"/>
```
[the remark this answers is not in the document]
Comment (sjchmiela) [reply template-files/android/AndroidManifest.xml]: Those are sample IDs from AdMob setup instructions

> Note: If you haven't created an AdMob account and registered an app yet, now's a great time to do so. In a real app, it is important that you use your actual AdMob app ID, not the one listed above. If you're just looking to experiment with the SDK in a Hello World app, though, you can use the sample App ID shown above. ([Source](https://developers.google.com/admob/ios/quick-start#update_your_infoplist))

**Review by ide — APPROVED** [review]

**Comment by changLiuUNSW** [comment]
@sjchmiela @ide 
When is this change going to release ? 
We only have 7 days left for the IOS 13. Is that possible to release this one as soon as possible. Otherwise my app will get blocked due to https://github.com/expo/expo/issues/5242

**Comment by ide** [comment]
We are looking to release SDK 35 before the iOS 13 public release (Sept 19).

**Comment by MehmetKaplan** [comment]
Hi,

First pardon me for the below dump question but I need a comment to decide if I really need to modify the code or not...

According to the SDK 35 release blog ( https://blog.expo.io/expo-sdk-35-is-now-available-beee0dfafbf4 ), this change seems as a breaking change **for all** Admob related codes. But when I test existing application without the app.json addition it seems fine.

> Admob: all Admob ads now require a new configuration value in app.json, expo.[platform].config.googleMobileAdsAppId. The value can be found by following the guide in this Google Support answer. (Related PR.)

So the question is, is it really required for all Admob ads or is it related with firebase only (#5242 and #5330)

**Comment by ide** [comment]
You need this for standalone apps. I would follow the instructions because even if things happen to work without changes, accidental side effects like that are not part of the public interface and might break whenever.

**Pull request merged**